Bottom: 182dccf75f29cf651863be00a595290413b2104f
Top:    35ecef1d8c84520d1606afa71a372901bb3f5dee
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-03-05 16:26:00 +0000

Object3D / step changes


---

diff --git a/src/hid/step/step_writer.c b/src/hid/step/step_writer.c
index dcd0472..9655d8f 100644
--- a/src/hid/step/step_writer.c
+++ b/src/hid/step/step_writer.c
@@ -22,7 +22,10 @@ fprint_id_list (FILE *f, step_id_list list)
        iter = g_list_next (iter)) {
     fprintf (f, "#%i, ", GPOINTER_TO_INT (iter->data));
   }
-  fprintf (f, "#%i )", GPOINTER_TO_INT (iter->data));
+  if (iter == NULL)
+    fprintf (f, ")");
+  else
+    fprintf (f, "#%i )", GPOINTER_TO_INT (iter->data));
 }
 
 static void
@@ -31,6 +34,12 @@ destroy_step_id_list (step_id_list list)
   g_list_free (list);
 }
 
+step_id_list
+step_id_list_append (step_id_list list, step_id id)
+{
+  return g_list_append (list, GINT_TO_POINTER (id));
+}
+
 /* NB: The GList this produces will leak if not passed back to one of the step_* functions which uses the list and destroys it */
 step_id_list
 make_step_id_list (int count, ...)
diff --git a/src/hid/step/step_writer.h b/src/hid/step/step_writer.h
index 1490bb4..c86aa1e 100644
--- a/src/hid/step/step_writer.h
+++ b/src/hid/step/step_writer.h
@@ -8,6 +8,7 @@ typedef struct {
 
 } step_file;
 
+step_id_list step_id_list_append (step_id_list list, step_id id);
 step_id_list make_step_id_list (int count, ...);
 
 step_file *step_output_file (FILE *f);
